src/components/stackAggregator/index.js

```diff
@@ -101,55 +114,81 @@ export default class StackAggregator extends PureBaseComponent {
     return 1;
   }
 
-  animate = () => {
-    this.animateValues();
-    this.animateCards();
+  animate = async () => {
```

Comment:
What is the reason you wrapped all animations with Promises?

Reply:
@Inbal-Tish so that if we want to - we can use the `onCollapseChanged` method which fires when the animation is complete
the animations only run synchronously if you provide the onCollapsedChanged prop, essentially not affecting existing functionality

Reply:
See the different way's animate is called here:

https://github.com/wix/react-native-ui-lib/pull/942/files/e3cd28520b4d50aa8adffb64d9fd1944f04defe9#diff-be29b451328dfdcd75825593d1e66f41R186-R189

it is await'ed if the `onCollapseWillChange` is a prop
it will just run it without awaiting if `onCollapseWillChange` is not defined in props

Reply:
Instead, You can provide the Animated.start() a callback for when the animating finishes where you can invoke the 'onCollapseChanged'. It looks something like this: 
`.start(({finished}) => {
    if (finished) {
      _.invoke(this.props, 'onCollapseChanged', this.state.collapsed);
    }
  });`

Reply:
@Inbal-Tish that is true, but to get my function to work as expected we need to make the animate function asynchronous.  In your example, it is synchronous.

See here:
https://github.com/wix/react-native-ui-lib/pull/942/files/e3cd28520b4d50aa8adffb64d9fd1944f04defe9#diff-d8bcdbc9df02b6a0cada95f9c4aa933158095b9369dc916e5ecc22468605b40aR171-R176